### PR TITLE
Fixes Cyborg miners not recieving the lavaproof trait on upgrade.

### PR DESCRIPTION
**Pull request by Vizoplez**
## About The Pull Request

Fixes broken code.

## Why It's Good For The Game
This somehow broke, and it seems to be because the trait is applied to the board itself, and not the cyborg.

As to why it is good, it allows this upgrade to finally work again (?)

## Changelog

:cl:
fix: Lavaproof upgrade now works.
/:cl:

### CODE REVIEW

**Comment by Mooshimi** [comment]
Have you tested this as a cyborg at all?

**Comment by Yawet330** [comment]
> Have you tested this as a cyborg at all?

will test now, but this was a phone edit. This bug however can be verified as this line of code via trait checking the board once installed.

**Comment by Yawet330** [comment]
image

image

Works fine.

**Comment by Mooshimi** [comment]
> Works fine.

Sweet!

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -277,12 +277,12 @@
 /obj/item/borg/upgrade/lavaproof/action(mob/living/silicon/robot/R, user = usr)
 	. = ..()
 	if(.)
-		ADD_TRAIT(src, TRAIT_LAVA_IMMUNE, type)
+		ADD_TRAIT(R, TRAIT_LAVA_IMMUNE, type)
```
Comment (Watermelon914) [inline code/game/objects/items/robot/robot_upgrades.dm:280]: make this variable name more detailed

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**